### ISSUE TITLE
TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by gossi** [emberwatch/ember-cli-typescript#38]
I do get this error fairly often:

```
$ ember s
Livereload server on http://localhost:49153
The Broccoli Plugin: [broccoli-typescript-compiler] failed with:
TypeError: Cannot read property 'length' of undefined
    at createSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:15396:109)
    at parseSourceFileWorker (/path/to/my/project/node_modules/typescript/lib/typescript.js:15328:26)
    at Object.parseSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:15277:26)
    at Object.createSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:15131:29)
    at SourceCache.getSourceFile (/path/to/my/project/node_modules/broccoli-typescript-compiler/dist/plugin.js:98:43)
    at Object.getSourceFile (/path/to/my/project/node_modules/broccoli-typescript-compiler/dist/plugin.js:300:85)
    at findSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:67716:29)
    at processSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:67647:27)
    at /path/to/my/project/node_modules/typescript/lib/typescript.js:67757:17
    at Object.forEach (/path/to/my/project/node_modules/typescript/lib/typescript.js:1443:30)
    at processReferencedFiles (/path/to/my/project/node_modules/typescript/lib/typescript.js:67755:16)
    at findSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:67740:21)
    at processSourceFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:67647:27)
    at processRootFile (/path/to/my/project/node_modules/typescript/lib/typescript.js:67535:13)
    at Object.createProgram (/path/to/my/project/node_modules/typescript/lib/typescript.js:66845:21)
    at Compiler.createProgram (/path/to/my/project/node_modules/broccoli-typescript-compiler/dist/plugin.js:236:22)

The broccoli plugin was instantiated at:
    at TypeScript.Plugin (/path/to/my/project/node_modules/broccoli-plugin/index.js:7:31)
    at new TypeScript (/path/to/my/project/node_modules/broccoli-typescript-compiler/dist/plugin.js:324:27)
    at typescript (/path/to/my/project/node_modules/broccoli-typescript-compiler/index.js:28:10)
    at TypeScriptPreprocessor.toTree (/path/to/my/project/node_modules/ember-cli-typescript/lib/typescript-preprocessor.js:96:7)
    at /path/to/my/project/node_modules/ember-cli-preprocess-registry/preprocessors.js:180:26
    at Array.forEach (native)
    at processPlugins (/path/to/my/project/node_modules/ember-cli-preprocess-registry/preprocessors.js:178:11)
    at module.exports.preprocessJs (/path/to/my/project/node_modules/ember-cli-preprocess-registry/preprocessors.js:171:10)
    at EmberApp.appAndDependencies (/path/to/my/project/node_modules/ember-cli/lib/broccoli/ember-app.js:1110:27)
    at EmberApp.javascript (/path/to/my/project/node_modules/ember-cli/lib/broccoli/ember-app.js:1225:30)
    at EmberApp.toArray (/path/to/my/project/node_modules/ember-cli/lib/broccoli/ember-app.js:1662:12)
    at EmberApp.toTree (/path/to/my/project/node_modules/ember-cli/lib/broccoli/ember-app.js:1684:38)
    at module.exports (/path/to/my/project/ember-cli-build.js:29:13)
    at Builder.setupBroccoliBuilder (/path/to/my/project/node_modules/ember-cli/lib/models/builder.js:56:19)
    at new Builder (/path/to/my/project/node_modules/ember-cli/lib/models/builder.js:30:10)
    at ServeTask.run (/path/to/my/project/node_modules/ember-cli/lib/tasks/serve.js:24:55)
```

 I'm trying to use typescript, however this pops up now and then. Here is what I do to solve this:

1) `rm -rf tmp/`
2) `rm -rf node_modules/ && yarn install`
3) `yarn remove ember-cli-typescript` `ember install ember-cli-typescript`

If it pops up too often, I cancel on typescript and keep it deinstalled (however, yes, I'm really trying to use it)

**Comment by pixelhandler** [comment]
@gossi I noticed this error today as well. I have a fresh install of ember-cli-typescript and am running ember-cli@2.11.0 on Windows with Node v6.11.0.

Are you running on MacOS, Windows, Linux? What version of Node.js, and ember-cli?

**Comment by pixelhandler** [comment]
@gossi just curious do use use `lib` in your `.tsconfig`? e.g. `lib: ["2017", "dom"]` for some reason I used this to support `Array.includes` but once I removed that from my config. (And converted to `.indexOf` instead.) I no longer had this error.

**Comment by gossi** [comment]
My configuration:

- No `lib`
- target: `ES2017`
- ember-cli: `2.13`
- node: `v7`
- platform: `macOS`

**Comment by chriskrycho** [comment]
@gossi also, what versions of TypeScript and ember-cli-typescript are you using? I'll try to dig into this sometime in the next couple weeks (though getting some motion on our typings story is a higher priority at present).

**Comment by gossi** [comment]
The version that was present at may 19th ;) Should be `v0.4.0` and that typescript version that comes with it. I can test it again next week, ping me in slack.

**Comment by chriskrycho** [comment]
Heh, saw the same error today while doing an upgrade. I'm even *more* strongly incentivized to track this down.

**Comment by chriskrycho** [comment]
Initial investigation is… perplexing. For me, it's throwing this error on a TypeScript-internal file: `/lib.dom.d.ts`, pulled in via the `lib` argument in the `tsconfig.json` file. 😬 Will update more later as I'm able to track it down further.

**Comment by chriskrycho** [comment]
I *believe* that a new version from the latest broccoli-typescript-compiler will solve this – will wait to see when that publishes, but initial work locally looks like that might do the trick.

**Comment by chriskrycho** [comment]
@gossi @pixelhandler please try it with the latest version on [this branch](https://github.com/emberwatch/ember-cli-typescript/tree/latest-b-c-ts):

```
"ember-cli-typescript": "https://github.com/emberwatch/ember-cli-typescript.git#b740a89d1f42461f4f59ec97d37bdd6608003337",
```

**Comment by chriskrycho** [comment]
Update: @gossi and @pixelhandler I fixed another bug on that branch this morning; maybe try the head of that branch instead:

```
"ember-cli-typescript": "https://github.com/emberwatch/ember-cli-typescript.git#b6c1a6f1e6dc9e8a0ef493f6a284df4f2d90e2ca
```

**Comment by billheaton-q2** [comment]
@chriskrycho I tried out the above branch and ran a build. twice in a row I had a cleanup error:

```
Cleanup error.
ENOTEMPTY: directory not empty, rmdir 'C:\...pathtorepo....\tmp\broccoli_merge_trees-output_path-siW7acsK.tmp'
```
Running Windows 10, Node v6.11

**Comment by pixelhandler** [comment]
@chriskrycho I tried out the above branch and ran a build. twice in a row I had a cleanup error:

```
Cleanup error.
ENOTEMPTY: directory not empty, rmdir 'C:\...pathtorepo....\tmp\broccoli_merge_trees-output_path-siW7acsK.tmp'
```

The build did work, was only a cleanup error. 

Running Windows 10, Node v6.11

**Comment by chriskrycho** [comment]
This should be resolved in the 1.0 beta. Please test with `"ember-cli-typescript": "next"` and let me know if it recurs.
